### PR TITLE
Styling part 1

### PR DESCRIPTION
**Pull request by mplorentz**
I'm going through and updating screens to match [Figma](https://www.figma.com/file/s0qf4VmyQygydP8MIQazZc/Nos?node-id=47%3A2&t=RoVj0lspTNDSLwlP-1), mostly. I'm taking some shortcuts because the stock UI components don't let us do everything Sebastian has designed. This is the first of a few PRs to do that and hopefully the biggest. 

I started reconfiguring the Discover tab's navigation bar to match Figma but then needed to 
- fix the double navigation bar (#83). For this I made some changed to Router and the side menu. The router now keeps track of separate paths for each tab, as well as the state of the side menu.
- fixed crashes #53, #49, and #100
- Added the new app icon #93
- Move the New Note button from the navigation bar into the tab bar
- Styled the home and discover feed so that they more or less match Figma
- Fixed SwiftUI previews for several views.

### CODE REVIEW

File: Nos/Views/NoteButton.swift [hunk 1]
```diff
@@ -15,20 +15,17 @@ import SwiftUI
...
     @EnvironmentObject private var relayService: RelayService
 
     var body: some View {
         if let author = note.author {
             Button {
-                // If we are on the home feed or this is not the root note, allow clicks on replies
-                if router.path.count == 0 || note.eventReferences?.count ?? 0 > 0
-                    || router.navigationTitle == Localized.profile.rawValue {
-                    router.path.append(note)
-                }
```
Comment (shanecb) [inline Nos/Views/NoteButton.swift:29]: I think this logic was protecting us from being able to open duplicate thread views for the same root note

File: Nos/Views/SettingsView.swift [hunk 2]
```diff
@@ -38,6 +39,7 @@ struct SettingsView: View {
                     if privateKeyString.isEmpty {
                         self.keyPair = nil
                         analytics.logout()
+                        appController.configureCurrentState()
```
Comment (shanecb) [inline Nos/Views/SettingsView.swift:42]: I like that this enables us to jump back out to onboarding. I don't love that when we complete onboarding again we land back on the settings page

File: Nos/Views/SideMenu.swift [hunk 3]
```diff
@@ -7,30 +7,44 @@
...
-        HStack {
-            SideMenuContent(closeMenu: closeMenu)
-                .frame(width: width, height: UIScreen.main.bounds.height)
-                .offset(x: menuOpened ? 0 : -width, y: -0.015 * UIScreen.main.bounds.height)
-                .animation(.default, value: menuOpened)
-            Spacer()
+            .transition(.move(edge: .leading))
+            // weirdly fixes dismissal animation
+            // https://sarunw.com/posts/how-to-fix-zstack-transition-animation-in-swiftui/#solution
+            .zIndex(1)
```
Comment (shanecb) [inline Nos/Views/SideMenu.swift:47]: huh, cool

File: Nos/Views/SideMenuContent.swift [hunk 4]
```diff
@@ -19,18 +19,15 @@ struct SideMenuContent: View {
...
+                Spacer()
                 HStack {
                     Button {
                         do {
                             guard let keyPair = KeyPair.loadFromKeychain() else { return }
                             let author = try Author.findOrCreate(by: keyPair.publicKeyHex, context: viewContext)
-                            router.path.append(author)
-                            router.navigationTitle = Localized.profile.string
-                            closeMenu()
+                            router.sideMenuPath.append(author)
```
Comment (shanecb) [inline Nos/Views/SideMenuContent.swift:30]: So we could use router.push here instead, right?

**Review by shanecb — APPROVED** [review]

File: Nos/Views/SettingsView.swift (hunk 2 above)
Comment (mplorentz) [reply Nos/Views/SettingsView.swift:42]: Good point. I think I can just clear the sideMenuPath and close the side menu right here. I will try that in my next PR.

File: Nos/Views/SideMenuContent.swift (hunk 4 above)
Comment (mplorentz) [reply Nos/Views/SideMenuContent.swift:30]: yeah, I didn't end up using push in all the places I could.

File: Nos/Views/NoteButton.swift (hunk 1 above)
Comment (mplorentz) [reply Nos/Views/NoteButton.swift:29]: will do in next PR.

**Pull request merged**